src/assets/scripts/aircraft/AircraftFlightManagementSystem.js

```diff
@@ -651,64 +688,24 @@ export default class AircraftFlightManagementSystem {
      *    - (spd) waypoint's speed restriction
      */
     climbViaSID() {
-        if (!this.currentLeg().type === FP_LEG_TYPE.SID) {
-            return;
+        if (this.currentLeg.type !== FP_LEG_TYPE.SID) {
+            return false;
         }
 
-        let wp = this.currentLeg().waypoints;
-        let cruise_alt = this.fp.altitude;
-        let cruise_spd = this.my_aircraft.model.speed.cruise;
+        const wp = this.currentLeg.waypoints;
+        const cruise_alt = this.fp.altitude;
+        const cruise_spd = this.my_aircraft.model.speed.cruise;
 
         for (let i = 0; i < wp.length; i++) {
-            let altitude = wp[i].fixRestrictions.alt;
-            let speed = wp[i].fixRestrictions.spd;
-            let minAlt;
-            let alt;
-            let maxAlt;
+            const waypoint = wp[i];
+            const { ctr_ceiling } = window.airportController.airport_get();
 
-            // Altitude Control
-            if (altitude) {
-                if (altitude.indexOf('+') !== -1) {
-                    // at-or-above altitude restriction
-                    minAlt = parseInt(altitude.replace('+', ''), 10) * 100;
-                    alt = Math.min(window.airportController.airport_get().ctr_ceiling, cruise_alt);
-                } else if (altitude.indexOf('-') !== -1) {
-                    maxAlt = parseInt(altitude.replace('-', ''), 10) * 100;
-                    // climb as high as restrictions permit
-                    alt = Math.min(maxAlt, cruise_alt);
-                } else {
-                     // cross AT this altitude
-                    alt = parseInt(altitude, 10) * 100;
-                }
-            } else {
-                alt = Math.min(window.airportController.airport_get().ctr_ceiling, cruise_alt);
-            }
-
-            wp[i].altitude = alt; // add altitudes to wp
-
-            let minSpd;
-            let spd = cruise_spd;
-            let maxSpd;
-            // Speed Control
-            if (speed) {
-                if (speed.indexOf('+') !== -1) {
-                    // at-or-above speed restriction
-                    minSpd = parseInt(speed.replace('+', ''), 10);
-                    spd = Math.min(minSpd, cruise_spd);
-                } else if (speed.indexOf('-') !== -1) {
-                    maxSpd = parseInt(speed.replace('-', ''), 10);
-                    // go as fast as restrictions permit
-                    spd = Math.min(maxSpd, cruise_spd);
-                } else {
-                     // cross AT this speed
-                    spd = parseInt(speed, 10);
-                }
-            }
-
-            // add speeds to wp
-            wp[i].speed = spd;
+            waypoint.setAltitude(ctr_ceiling, cruise_alt);
+            waypoint.setSpeed(cruise_spd);
```

Comment:
so now I have model methods doing this logic, but I feel they should be run when the waypoints are set not here after the fact.